src/rgw/rgw_putobj_processor.cc

```diff
@@ -142,26 +142,36 @@ RadosWriter::~RadosWriter()
    */
   for (const auto& obj : written) {
     if (raw_head && obj == *raw_head) {
-      ldout(store->ctx(), 5) << "NOTE: we should not process the head object (" << obj << ") here" << dendl;
+      ldpp_dout(this, 5) << "NOTE: we should not process the head object (" << obj << ") here" << dendl;
       need_to_remove_head = true;
       continue;
     }
 
     int r = store->delete_raw_obj(obj);
     if (r < 0 && r != -ENOENT) {
-      ldout(store->ctx(), 5) << "WARNING: failed to remove obj (" << obj << "), leaked" << dendl;
+      ldpp_dout(this, 5) << "WARNING: failed to remove obj (" << obj << "), leaked" << dendl;
     }
   }
 
   if (need_to_remove_head) {
-    ldout(store->ctx(), 5) << "NOTE: we are going to process the head obj (" << *raw_head << ")" << dendl;
+    ldpp_dout(this, 5) << "NOTE: we are going to process the head obj (" << *raw_head << ")" << dendl;
     int r = store->delete_obj(obj_ctx, bucket_info, head_obj, 0, 0);
     if (r < 0 && r != -ENOENT) {
-      ldout(store->ctx(), 0) << "WARNING: failed to remove obj (" << *raw_head << "), leaked" << dendl;
+      ldpp_dout(this, 0) << "WARNING: failed to remove obj (" << *raw_head << "), leaked" << dendl;
     }
   }
 }
 
+unsigned RadosWriter::get_subsys() const
+{
+  return dout_subsys;
+}
+
+std::ostream& RadosWriter::gen_prefix(std::ostream& out) const
+{
+  return out << "Put Obj Processor: ";
```

Comment:
this putobj processor code is called by ops, which already have a DoutPrefixProvider. the way you've structured this, it throws out any existing prefixes from req_state/RGWOp and only uses this prefix

i see two options here..
* drop these 'Obj Processor' prefixes and just pass in a dpp from the calling RGWOp, or
* change these classes to use DoutPrefixPipe instead, which takes the existing dpp from RGWOp, and overload add_prefix() to add on to its prefix

since the main caller will be RGWPutObj, its log output will already look like this:
> req 484 0.002s s3:put_obj ...

so i'm not sure that any extra putobj-related prefix is necessary here. what do you think?

Reply:
@cbodley Can you give me an example of where the putobj processor code is being called by the ops? Files to look at and what to search for would be helpful too.

Reply:
in rgw_op.cc, `RGWPutObj::execute()` puts a `processor` on the stack, which is either `MultipartObjectProcessor` or `AtomicObjectProcessor`. it calls `processor->prepare()` to initialize, feeds it all the data with several calls to `filter->process(data)`, then finishes with a call to `processor->complete()`

a grep for ObjectProcessor should catch the other uses, which should look pretty similar